airflow-core/tests/unit/models/test_deadline.py

```diff
@@ -371,11 +373,25 @@ def test_deadline_database_integration(self, reference, expected_column, session
             if expected_column is not None:
                 result = reference.evaluate_with(session=session, interval=interval, **conditions)
                 mock_fetch.assert_called_once_with(expected_column, session=session, **conditions)
+            elif reference == DeadlineReference.AVERAGE_RUNTIME:
+                with mock.patch("airflow._shared.timezones.timezone.utcnow") as mock_utcnow:
+                    mock_utcnow.return_value = DEFAULT_DATE
+                    with mock.patch.object(session, "execute") as mock_execute:
```

Comment:
It looks like this a DB test so I'm not sure if we should be mocking session.execute